test/parallel_api/algorithm/alg.sorting/sort.pass.cpp

```diff
@@ -174,82 +174,237 @@ Equal(std::int32_t x, std::int32_t y)
     return x == y;
 }
 
+template <typename InputIterator, typename OutputIterator1, typename OutputIterator2, typename Size>
+void
+copy_data(InputIterator first, OutputIterator1 expected_first, OutputIterator1 expected_last, OutputIterator2 tmp_first,
+          Size n)
+{
+    ::std::copy_n(first, n, expected_first);
+    ::std::copy_n(first, n, tmp_first);
+}
+
+template <typename Iterator, typename Compare = ::std::less<typename std::iterator_traits<Iterator>::value_type>>
+void
+sort_data(Iterator __from, Iterator __to, Compare compare = {})
+{
+    if (Stable)
+        ::std::stable_sort(__from, __to, compare);
+    else
+        ::std::sort(__from, __to, compare);
+}
+
+template <typename Policy, typename Iterator,
+          typename Compare = ::std::less<typename std::iterator_traits<Iterator>::value_type>>
+void
+sort_data(Policy&& exec, Iterator __from, Iterator __to, Compare compare = {})
+{
+    if (Stable)
+        ::std::stable_sort(exec, __from, __to, compare);
+    else
+        ::std::sort(exec, __from, __to, compare);
+}
+
+template <typename OutputIterator1, typename OutputIterator2, typename Size>
+void
+check_results(OutputIterator1 expected_first, OutputIterator2 tmp_first, Size n, const char* error_msg)
+{
+    for (size_t i = 0; i < n; ++i, ++expected_first, ++tmp_first)
+    {
+        // Check that expected[i] is equal to tmp[i]
+        EXPECT_TRUE(Equal(*expected_first, *tmp_first), error_msg);
+    }
+}
+
 template <typename T>
 struct test_sort_with_compare
 {
+#if TEST_DPCPP_BACKEND_PRESENT
+#if _PSTL_SYCL_TEST_USM
+    template <sycl::usm::alloc alloc_type, typename Policy, typename InputIterator, typename OutputIterator,
+              typename OutputIterator2, typename Size, typename Compare>
+    typename ::std::enable_if<TestUtils::is_base_of_iterator_category<::std::random_access_iterator_tag, InputIterator>::value,
+                              void>::type
+    test_usm(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
```

Comment:
Fixed by adding test_sort_base and moving to this structure some common functions

Reply:
> Looking at it one more time, I guess we can keep just one testing functor instead of two - test_sort_with_compare and test_sort_without_compare. Indeed, please have a look an example https://godbolt.org/z/rrPdMbYnK

I think that this proposal is not applicable in this case, because methods of test_sort_with_compare and methods of test_sort_without_compare has different SFINAE conditions.

Reply:
> > Looking at it one more time, I guess we can keep just one testing functor instead of two - test_sort_with_compare and test_sort_without_compare. Indeed, please have a look an example https://godbolt.org/z/rrPdMbYnK
> 
> I think that your proposal is not applicable in this case, because methods of test_sort_with_compare and methods of test_sort_without_compare has different SFINAE conditions.

I don't think so. Please, have a look https://github.com/oneapi-src/oneDPL/blob/dev/test_sort_usm_device/test/parallel_api/algorithm/alg.sorting/sort.pass.cpp#L194